BaseStation/VideoStreamer/MainWindow.xaml.cs

```diff
@@ -1,29 +1,231 @@
+// capabilities.
+// Gstreamer must be installed to the C drive
+// TO DO:
+// find a final loaction for ports file and create generic destination to work on any computer
+
+// server command:
+// IP will cange when switching to basestation computer
+// gst-launch-1.0 v4l2src device="/dev/video0" ! "video/x-raw, format=(string)I420, width=(int)1280, height=(int)720" ! omxh264enc ! 'video/x-h264, stream-format=(string)byte-stream' ! h264parse ! rtph264pay ! udpsink host=192.168.0.5 port=5555
+// client test:
+// gst-launch-1.0 -vvv -e udpsrc caps=\"application/x-rtp, media=(string)video, clock-rate=(int)90000, encoding-name=(string)H264\" port=5555 ! rtph264depay ! h264parse ! mp4mux ! filesink location=test.mp4
 namespace VideoStreamer
 {
-    /// <summary>
-    /// Interaction logic for MainWindow.xaml
-    /// </summary>
-    public partial class MainWindow : Window
+    public partial class MainWindow : Window, INotifyPropertyChanged
     {
+        private DispatcherTimer recordingTime;
+        private DateTime duration;
+        private String portFileLocation;
+        private String _destination;
+        public String Destination
+        {
+            get { return _destination; }
+            set
+            {
+                _destination = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs("Destinaion"));
+            }
+        }
+        private Boolean playing;
+
+        private ObservableCollection<int> ports;
+        private Process[] players;
+        private Process[] recorders;
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public ObservableCollection<int> Ports
+        {
+            get => ports;
+        }
+
         public MainWindow()
         {
+            recordingTime = new DispatcherTimer();
+            recordingTime.Tick += new EventHandler(dispatcherTimer_Tick);
+            recordingTime.Interval = new TimeSpan(0, 0, 1);
+            duration = new DateTime(0);
+            DataContext = this;
+            Destination = @"C:\Users\emela\Desktop\";
+            // I know this is bad but I don't know where it should go
+            portFileLocation = Directory.GetCurrentDirectory() + @"\ports.txt";
+
+            ports = new ObservableCollection<int>();
+            initPorts();
+
             InitializeComponent();
         }
+
+        // initializes the ports from the portfilelocation
+        private void initPorts()
+        {
+            String line;
+            int x;
+            StreamReader file = new StreamReader(portFileLocation);
+            while ((line = file.ReadLine()) != null)
+            {
+                if (Int32.TryParse(line, out x))
+                {
+                    ports.Add(x);
+                }
+            }
+            file.Close();
+        }
+
+        // initializes the processes used to launch the stream and file recorder
+        private void initProcesses(ref Process[] pro)
```

Comment:
Make this into a method called makeShellProcess, which returns a correctly configured process. Then you can add these correctly configured processes to the array (or list) as needed.